takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/DeployParticipant.java

```diff
@@ -0,0 +1,102 @@
+package io.takari.maven.plugins;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.AbstractMavenLifecycleParticipant;
+import org.apache.maven.Maven;
+import org.apache.maven.MavenExecutionException;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.execution.MavenSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.deployment.DeployRequest;
+import org.eclipse.aether.deployment.DeploymentException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Named
+@Singleton
+public class DeployParticipant extends AbstractMavenLifecycleParticipant {
+
+  private static final Logger log = LoggerFactory.getLogger(AbstractMavenLifecycleParticipant.class);
+
+  protected RepositorySystem repoSystem;
+  private List<DeployRequest> deployAtEndRequests = Collections.synchronizedList(new ArrayList<DeployRequest>());
+
+  @Inject
+  public DeployParticipant(RepositorySystem repoSystem) {
+    this.repoSystem = repoSystem;
+  }
+
+  @Override
+  public void afterSessionEnd(MavenSession session) throws MavenExecutionException {
+    boolean noerrors = session.getResult().getExceptions().isEmpty();
+
+    if (!deployAtEndRequests.isEmpty() && noerrors) {
+      log.info("");
+      log.info("------------------------------------------------------------------------");
+      log.info("-- Performing deploy at end                                           --");
+      log.info("------------------------------------------------------------------------");
+
+      synchronized (deployAtEndRequests) {
+        for (DeployRequest deployRequest : deployAtEndRequests) {
+          try {
+            deploy(session.getRepositorySession(), deployRequest);
+          } catch (DeploymentException e) {
+            log.error(e.getMessage(), e);
+            throw new MavenExecutionException(e.getMessage(), e);
+          }
+        }
+        deployAtEndRequests.clear();
+      }
+
+      log.info("------------------------------------------------------------------------");
+    }
+  }
+
+  public List<DeployRequest> getDeployAtEndRequests() {
```

Comment:
Would need to move it to the same package as the test. Ok with *.install_deploy?